sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1179,55 +1114,40 @@ public void close() throws IOException {
      * Returns a new {@link Write} transform that write to given topic.
      */
     public Write<K, V> withTopic(String topic) {
-      return new Write<K, V>(topic, keyCoder, valueCoder, producerConfig);
+      return toBuilder().setTopic(topic).build();
     }
 
     /**
      * Returns a new {@link Write} with {@link Coder} for serializing key (if any) to bytes.
      * A key is optional while writing to Kafka. Note when a key is set, its hash is used to
      * determine partition in Kafka (see {@link ProducerRecord} for more details).
      */
+    @SuppressWarnings({"unchecked", "rawtypes"})
```

Comment:
As we discussed, please check with Dan about requiring user to set the type initially (```KafkaIO.<String, byte[]>Read()...```).